solidity/test/isms/CCIPIsm.t.sol

```diff
@@ -0,0 +1,364 @@
+        deployCCIPHookAndISMOptimism();
+        deployCCIPHookAndISMMainnet();
+
+        vm.selectFork(optimismFork);
+        ccipISMOptimism.setAuthorizedHook(
+            TypeCasts.addressToBytes32(address(ccipHookMainnet))
+        );
+
+        // for sending value
+        vm.deal(OP_ROUTER_ADDRESS, 2 ** 255);
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                         FORK TESTS                          ///
+    ///////////////////////////////////////////////////////////////////
+
+    /* ============ hook.quoteDispatch ============ */
+
+    function testFork_quoteDispatch() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        testMetadata = _buildMetadata(
+            OPTIMISM_CHAIN_SELECTOR,
+            address(ccipISMOptimism)
+        );
+
+        assert(ccipHookMainnet.quoteDispatch(testMetadata, encodedMessage) > 0);
+    }
+
+    /* ============ hook.postDispatch ============ */
+    event Refund(uint256 amount);
+
+    receive() external payable {
+        emit Refund(msg.value);
+    }
+
+    function testFork_postDispatch() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        bytes memory encodedHookData = abi.encode(messageId);
+
+        l1Mailbox.updateLatestDispatchedId(messageId);
+
+        uint256 quotedFee = ccipHookMainnet.quoteDispatch(
+            testMetadata,
+            encodedMessage
+        );
+
+        ccipHookMainnet.postDispatch{value: (quotedFee * 11) / 10}(
+            testMetadata,
+            encodedMessage
+        );
+    }
+
+    function testFork_postDispatch_RevertWhen_NotEnoughValueSent() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        l1Mailbox.updateLatestDispatchedId(messageId);
+
+        vm.expectRevert();
+
+        ccipHookMainnet.postDispatch(testMetadata, encodedMessage);
+    }
+
+    function testFork_postDispatch_RevertWhen_NotLastDispatchedMessage()
+        public
+    {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        vm.expectRevert(
+            "AbstractMessageIdAuthHook: message not latest dispatched"
+        );
+
+        ccipHookMainnet.postDispatch(testMetadata, encodedMessage);
+    }
+
+    // /* ============ ISM.verifyMessageId ============ */
+
+    function testFork_verifyMessageId() public {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        Client.Any2EVMMessage memory message = _encodeCCIPReceiveMessage();
+
+        vm.prank(OP_ROUTER_ADDRESS);
+        ccipISMOptimism.ccipReceive(message);
+
+        assertTrue(ccipISMOptimism.verifiedMessages(messageId).isBitSet(255));
+    }
+
+    function testFork_verifyMessageId_RevertWhen_SenderNotAllowed() public {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        Client.Any2EVMMessage memory message = _encodeCCIPReceiveMessage();
+        message.sender[0] = ~message.sender[0];
```

Comment:
Ditto again make this a function argument

Reply:
we use this pattern elsewhere